docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+import argparse
+import os
+import re
+import shutil
+from argparse import BooleanOptionalAction
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+MD_HEADER_REGEX = (
+    r"^"            # start of line
+    r"(\s{0,3})"    # Group 1: zero to three whitespaces, tabs etc (four and more spaces are no heading)
+    r"(#{1,6})"     # Group 2: one or up to 6 '#'
+    r"(\s{0,})"     # Group 3: zero or more whitespaces
+    r"([^#].*)"     # Group 4: one or more chars but not starting with another #         
```

Comment:
Nit: Maybe remove the trailing spaces?

Reply:
Do you mean the spaced between the parts of the regex and the comment? I believe it's more accessible this way but am open to discuss.

Reply:
I'll resolve this discussion and merge the PR. Please reach out if you still see the need for a change.